bpf/generictracer/protocol_http.h

```diff
@@ -401,13 +402,100 @@ static __always_inline void handle_http_response(unsigned char *small_buf,
     cleanup_http_request_data(pid_conn, info);
 }
 
-// k_tail_protocol_http
-SEC("kprobe/http")
-int obi_protocol_http(void *ctx) {
+static __always_inline int __obi_continue2_protocol_http(struct pt_regs *ctx,
+                                                         call_protocol_args_t *args,
+                                                         http_info_t *info,
+                                                         http_connection_metadata_t *meta) {
     (void)ctx;
 
+    if (meta) {
+        u32 type = trace_type_from_meta(meta);
+        tp_info_pid_t *tp_p = trace_info_for_connection(&args->pid_conn.conn, type);
+        if (tp_p) {
+            info->tp = tp_p->tp;
+            if (args->self_ref_parent_id) {
```

Comment:
Hey @mmat11, this self_reference parent id seems unrelated to this change, does it make sense to have this as a separate PR? Or maybe I misunderstand the code?

Reply:
this is needed because of the newly added tail calls, the ID needs to get resolved at the entrypoint and processed at the end so by storing it in args I didn't change the current logic